File: leetcode/python/2530.maximal-score-after-applying-k-operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        maxHeap = [-n for n in nums]
        heapq.heapify(maxHeap)

        ans = 0

        for _ in range(k):
            maxNum = heapq.heappop(maxHeap)
            ans -= maxNum
            heapq.heappush(maxHeap, maxNum//3)
        return ans

File: leetcode/cpp/2530.maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int> maxHeap;
        for (const int n: nums) {
            maxHeap.emplace(n);
        }
        long ans = 0;
        for (int i=0; i < k; ++i) {
            int maxNum = maxHeap.top();
            maxHeap.pop();
            ans += maxNum;
            maxHeap.emplace((maxNum+2)/3);
        }
        return ans;
    }
};

